homeassistant/components/fronius/sensor.py

```diff
@@ -657,8 +671,10 @@ def _handle_coordinator_update(self) -> None:
         """Handle updated data from the coordinator."""
         try:
             self._attr_native_value = self._get_entity_value()
-        except KeyError:
+        except (KeyError, ValueError):
             # sets state to `None` if no default_value is defined in entity description
+            # KeyError: raised when omitted in response - eg. at night when no production
+            # ValueError: raised when invalid Zero value received
             self._attr_native_value = self.entity_description.default_value
         self.async_write_ha_state()
```

Comment:
This will make it unknown rather than unavailable. Which probably is fine here. Maybe call the flag invalid_when_zero?

Reply:
Oh, right thanks 👍. I did confuse these states 😬

Reply:
Another way would be to just return without changing `_attr_native_value` and calling `async_write_ha_state` - which would just preserve the previous state. Would you prefer that over "unknown"?

Reply:
Would mean different behavior on startup of ha compared to when running (unless we restore state), so i think going to unknown ls good enough.